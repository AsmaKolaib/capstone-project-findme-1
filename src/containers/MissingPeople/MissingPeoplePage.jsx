import React, { useState, useEffect, useRef } from 'react';
// import { CircleArrow as ScrollUpButton } from 'react-scroll-up-button';
import MissingPersonCard from '../../components/MissingPersonCard/MissingPersonCard';
import Model from 'react-modal'
import { ReactComponent as MyIcon } from './repeat-grid-4.svg';
import Search from '../../components/Search/Search'
import './style.css';
import NiceButton from '../../components/LoadingButton/NiceButton';
import '../../components/LoadingButton/style.css';
import useFirestore from '../../hooks/useFirestore'
import { Link, Route } from 'react-router-dom';
import { getSearchInfo } from '../../components/Search/searchInfo'
import {getCurrentDate} from '../../components/Search/helpers'
const MissingPeoplePage = ({searchInfo, searched}) => {
  const ITEMSTOSHOW = 8;
  const { docs } = useFirestore('images');
  const [data, setData] = useState([]); // maybe set the initial value later
  const [visible, setVisible] = useState(ITEMSTOSHOW);
  const [buttonText, setButtonText] = useState('⬇ Show More');
  const buttonTextRef = useRef('⬇ Show More');
  const [matched, setMatched] = useState([]);
  const [isSecondButtonLoading, setIsSecondButtonLoading] = useState(false);
  const [loadingSpeed, setLoadingSpeed] = useState(1);
  const [isOpen, setIsOpen] = useState(false);

  const dynamicSearch =(entered)=>{

    if (searchInfo.goneMissingOn===getCurrentDate()&& !entered.isLookingForFamily  )
     return docs.filter(name=> name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase()))
    else if (searchInfo.goneMissingOn===getCurrentDate()&& entered.isLookingForFamily)
     return docs.filter(name=> name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase())
     && name.values.isLookingFor)
     else if (searchInfo.goneMissingOn!==getCurrentDate()&& !entered.isLookingForFamily)
     return docs.filter(name=>{ return name.values.lostSince.toLowerCase().includes(entered.goneMissingOn.toLowerCase()) && 
      name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase())
      
    })

     else 
     return docs.filter(name=>{  name.values.lostSince.toLowerCase().includes(entered.goneMissingOn.toLowerCase()) && 
      name.values.fristName.toLowerCase().includes(entered.missingName.toLowerCase()&& name.values.isLookingFor)
      
    })
    
    
    }
  useEffect(() => {
    if (isSecondButtonLoading) {
      setTimeout(() => {
        setIsSecondButtonLoading(false);
      }, 1000 / loadingSpeed);
    }
  }, [isSecondButtonLoading, loadingSpeed]);

  useEffect(() => {
    setData(() => docs);
  }, [docs]);
  const getData =()=>{

  }
  
  const showMoreItems = () => {
    if (visible > docs.length) {
      // setButtonText(() => 'complete');
      buttonTextRef.current = 'Complete';
    }

    // setButtonText(() => '...loading');
    // buttonTextRef.current = '..loading';
    setVisible((prevValue) => prevValue + ITEMSTOSHOW);
    // buttonTextRef.current = '⬇ Show More';
    // setButtonText(() => '⬇ Show More');
  };

  const showMoreItemsClick = () => {
    // buttonTextRef.current = '..loading';
    setTimeout(() => {
      showMoreItems();
    }, 1000);
    clearTimeout();
    // buttonTextRef.current = '⬇ Show More';
  };

  return (
    <div>

      <MyIcon className="absolute" />
      <MyIcon className="absolute right-0 mt-64 " />

      <div className="flex flex-wrap ml-8 xl:justify-center">
        {data.length === 0 ? (
          <h2 className="flex-auto text-center">not found results</h2>
        ) : (
          searched?(
           // dynamicSearch(docs, searchInfo)
           /*matched = docs && docs
            .filter((item, index) => (
             item.values.fristName.toLowerCase().includes(searchInfo.missingName.toLowerCase())

            ))*/
            dynamicSearch(searchInfo)
            .slice(0, visible)
            .map((item, index) => (
              //searchFun(item, searchInfo)==='nameSearch'?(
             // item.values.fristName.toLowerCase()===searchInfo.missingName ?
              <Link key={item.id}
              to={`/missing_people_details/${item.id}`}
                >
              <MissingPersonCard key={item.id}  id={item.id} cardInfo={item} i={index} />
              </Link>


                 )     )
                  
            )
            : (
            docs && docs
            .slice(0, visible)
            .map((item, index) => (
              <Link key={item.id}
              to={`/missing_people_details/${item.id}`}
                >
              <MissingPersonCard key={item.id}  id={item.id} cardInfo={item} i={index} />
              </Link>
            ))
            
        ))}
      </div>
      {console.log(visible, data.length)}
      <div className=" ">
        {visible > docs.length || visible > dynamicSearch(searchInfo).length ?  (
          ''
        ) : (
          <NiceButton
            className="focus:outline-none mx-auto my-10 py-2 px-4 align-center text-blue-500 border-solid border-2 border-blue-500 rounded-3xl hover:text-blue-600 hover:border-blue-600 block font-light "
            isLoading={isSecondButtonLoading}
            onClick={() => {
              setIsSecondButtonLoading(true);
              showMoreItemsClick();
            }}
          >
            ⬇ Show More
          </NiceButton>
        )}
      </div>
    </div>
  );
};

// const removeOutline = () => {};

export default MissingPeoplePage;

// const scrollButtonStyle = {
//   position: 'fixed',
//   right: '-100px',
//   bottom: '150px',
//   transition: 'right 0.5s',
//   cursor: 'pointer',
//   backgroundColor: '#2B6CB0',
//   color: 'white',
//   fontSize: '20px',
//   padding: '10px',
//   borderColor: '#2B6CB0ed', // #2B6CB0
//   border: 'none',
//   fill: 'white',
//   focus: 'focus:outline-none',
// };

// const scrollButtonTransition = {
//   right: '20px',
// };