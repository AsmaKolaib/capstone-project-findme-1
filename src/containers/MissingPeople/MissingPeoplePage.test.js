import React from 'react';
import ReactDOM from 'react-dom';
import { getByTestId, render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MissingPeoplePage from './MissingPeoplePage';

afterEach(cleanup);

it('render without crashes ', () => {
  const div = document.createElement('div');
  ReactDOM.render(<MissingPeoplePage />, div);
});

it('renders snapshot 1', () => {
  const tree = renderer.create(<MissingPeoplePage />).toJSON();
  expect(tree).toMatchSnapshot();
});
