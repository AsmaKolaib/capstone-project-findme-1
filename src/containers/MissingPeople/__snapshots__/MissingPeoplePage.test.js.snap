// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1 1`] = `
<div>
  <svg
    className="absolute"
  >
    repeat-grid-4.svg
  </svg>
  <svg
    className="absolute right-0 mt-64 "
  >
    repeat-grid-4.svg
  </svg>
  <div
    className="flex flex-wrap ml-8"
  >
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Found
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "profile-svg.svg",
              }
            }
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 6-6-2016
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-10 py-2 font-bold md:pr-2 md:pl-5 md:py-1 lg:pl-8 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Lost
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src="image.jpg"
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 7-7-2017
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Found
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src="image2.jpg"
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 8-8-2018
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-10 py-2 font-bold md:pr-2 md:pl-5 md:py-1 lg:pl-8 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Lost
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src="image3.png"
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 9-9-2019
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Found
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src="image4.jpg"
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 10-10-2020
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-6 py-2 font-bold md:pr-2 md:pl-3 md:py-1 lg:pl-6 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Found
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src="image5.jpg"
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 7-7-2017
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-10 py-2 font-bold md:pr-2 md:pl-5 md:py-1 lg:pl-8 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Lost
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src="image6.png"
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 7-7-2017
          </p>
        </div>
      </div>
    </div>
    <div
      className="rounded-3xl mx-5 my-3 mx-auto md:mx-4"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="shadow-lg rounded-3xl max-w-xs overflow sm:self-center sm:mx-auto max-h-full"
      >
        <div
          className="relative"
        >
          <span
            className="absolute inline-block pr-4 pl-10 py-2 font-bold md:pr-2 md:pl-5 md:py-1 lg:pl-8 lg-pr-4 lg:py-2 bg-blue-700 right-0 text-white rounded-l-3xl mt-8 md:mt-4 lg:mt-8"
          >
            Lost
          </span>
          <img
            alt="John Doe image"
            className="opacity-1 cursor-pointer h-64"
            src={
              Object {
                "ReactComponent": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "render": [Function],
                },
                "__esModule": true,
                "default": "profile-svg.svg",
              }
            }
            width={300}
          />
        </div>
        <div
          className="px-6 py-4 text-left"
        >
          <div
            className="font-bold text-xl mb-2"
          >
            John Doe
          </div>
          <p
            className="text-base font-light"
          >
            Lost since: 7-7-2017
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className=" "
  >
    <button
      className="button"
      onClick={[Function]}
      style={Object {}}
    >
      <div
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        â¬‡ Show More
      </div>
    </button>
  </div>
  <aside
    aria-label="Scroll to top of page"
    className="ScrollUpButton__Container "
    data-testid="react-scroll-up-button"
    onClick={[Function]}
    onKeyPress={[Function]}
    role="button"
    style={
      Object {
        "WebkitTransition": "all 0.5s ease-in-out",
        "backgroundColor": "#2B6CB0",
        "border": "none",
        "borderColor": "#2B6CB0ed",
        "borderRadius": "50%",
        "bottom": "150px",
        "color": "white",
        "cursor": "pointer",
        "fill": "white",
        "focus": "focus:outline-none",
        "fontSize": "20px",
        "height": 50,
        "opacity": 0,
        "padding": "10px",
        "position": "fixed",
        "right": "-100px",
        "transition": "right 0.5s",
        "transitionProperty": "opacity, right",
        "width": 50,
      }
    }
    tabIndex={-1}
  >
    <svg
      viewBox="0 0 32 32"
    >
      <path
        d="M27.414 12.586l-10-10c-0.781-0.781-2.047-0.781-2.828 0l-10 10c-0.781 0.781-0.781 2.047 0 2.828s2.047 0.781 2.828 0l6.586-6.586v19.172c0 1.105 0.895 2 2 2s2-0.895 2-2v-19.172l6.586 6.586c0.39 0.39 0.902 0.586 1.414 0.586s1.024-0.195 1.414-0.586c0.781-0.781 0.781-2.047 0-2.828z"
      />
    </svg>
  </aside>
</div>
`;
